### PR TITLE
Adding retry attempts in OSS repo workflows

### PR DESCRIPTION
**Pull request by Mykyta-Serbynevskyi**
As per https://github.com/airbytehq/airbyte-cloud/issues/3109 this PR adds retry attempts in GH workflows in `airbytehq/airbyte` repo.
The following retry actions were used:

https://github.com/marketplace/actions/retry-action (can be used for actions & steps)
https://github.com/marketplace/actions/retry-step (can be used for steps only)

### CODE REVIEW

File: .github/workflows/publish-command.yml [hunk 1]
```diff
@@ -288,18 +288,25 @@ jobs:
           echo "IMAGE_VERSION=$(_get_docker_image_version ${DOCKERFILE})" >> $GITHUB_ENV
       - name: Prepare Sentry
         if: startsWith(matrix.connector, 'connectors')
-        run: |
-          curl -sL https://sentry.io/get-cli/ | bash || echo "sentry cli already installed"
+        uses: nick-fields/retry@v2
+        with:
+          timeout_seconds: 15
```
Comment (git-phu) [inline .github/workflows/publish-command.yml:293]: would it be fine to retry without a timeout for this?

**Review by supertopher — APPROVED** [review]
this looks solid to me, I think we are safe to merge as long as you have seen this run once.

**Review by git-phu — COMMENTED** [review]
looks great!

just one question

File: .github/workflows/publish-command.yml (hunk 1 above)
Comment (Mykyta-Serbynevskyi) [reply .github/workflows/publish-command.yml]: Done, I have removed timeout for this.

**Comment by davydov-d** [comment]
Hey team, this PR brings in an unexpected behavior as for me. 
Looks like `fail-fast: false` + `retries` (correct me if I'm wrong) make long failing builds even longer. I bet we should only retry transient errors.
This one was killed due to a timeout (240 mins):
https://github.com/airbytehq/airbyte/actions/runs/3648925714/jobs/6162969779 - the reason is the tests are not passing.

Is it possible to alter the workflows so they would not run that long?

**Pull request merged**